manager/controllers/app/select_modules.go

```diff
@@ -346,7 +346,30 @@ func allowCapability(item *DataInfo, capability taxonomy.Capability) bool {
 }
 
 func validateModuleRestrictions(item *DataInfo, edge *Edge) bool {
-	// TODO(shlomitk1): validate module restrictions
+	capability := edge.Module.Spec.Capabilities[edge.CapabilityIndex]
+	return validateModuleRestrictionsPerCapability(item, edge, capability.Capability)
+}
+
+func validateModuleRestrictionsPerCapability(item *DataInfo, edge *Edge, capability taxonomy.Capability) bool {
+	restrictions := item.Configuration.ConfigDecisions[capability].DeploymentRestrictions.Modules
+	if len(restrictions) == 0 {
+		return true
+	}
+	moduleCapability := edge.Module.Spec.Capabilities[edge.CapabilityIndex]
+	capabilityDetails, err := utils.StructToMap(&moduleCapability)
+	if err != nil {
+		return false
+	}
+	for key, values := range restrictions {
+		fields := strings.Split(key, ".")
+		value, found, err := unstructured.NestedString(capabilityDetails, fields...)
```

Comment:
This won't work for a property that belongs to a module capability. For example, "capabilities.scope" does not really exist. For a path capabilities.XXX you should look for XXX inside a selected capability, or change to 
capabilities.<edge.CapabilityIndex>.XXX

Reply:
Thanks for the comment. I'd like to confirm that for the module restriction, it can be any property inside a moduleCapability or it is limited to Scope?

Reply:
This can be any property inside a capability, or in the module spec unrelated to the capabilities, e.g. "type"